app/services/arcgis_api/geocoder.rb

```diff
@@ -184,7 +171,16 @@ def handle_api_errors(response_body)
         # {"error"=>{"code"=>400, "message"=>"", "details"=>[""]}}
         error_code = response_body.dig('error', 'code')
         error_message = response_body.dig('error', 'message') || "Received error code #{error_code}"
-
+        # log an error
+        analytics.idv_arcgis_request_failure(
+          exception_class: 'ArcGIS',
+          exception_message: error_message,
+          response_body_present: false,
+          response_body: '',
+          response_status_code: error_code,
+          api_status_code: error_code,
+        )
+        Rails.cache.delete(API_TOKEN_CACHE_KEY) # this might only be needed for local testing
```

Comment:
@racingspider Can you inspect this cache clear more closely to verify if it's needed for these cases, and open a PR to remove it if not?